api/cloud-control/v1beta1/network_types.go

```diff
@@ -0,0 +1,187 @@
+	// the Scope keeps track of the SKR that network was mentioned. The network reference has all parameters
+	// required for cloud provider access.
+	//
+	// +kubebuilder:validation:Required
+	Scope ScopeRef `json:"scope"`
+
+	// +kubebuilder:validation:Required
+	Network NetworkInfo `json:"network"`
+
+	// +kubebuilder:default=external
+	Type NetworkType `json:"type"`
+}
+
+// NetworkInfo can be one of ManagedNetwork or NetworkReference
+//
+// +kubebuilder:validation:MinProperties=1
+// +kubebuilder:validation:MaxProperties=1
+type NetworkInfo struct {
+	// Managed described the parameters of the network that will be created and managed by Cloud Manager.
+	// The deletion of the Netowork resource will
+	// If specified, then Reference must be nil.
+	// +optional
+	Managed *ManagedNetwork `json:"managed,omitempty"`
+
+	// Reference describes externally managed network that Cloud Manager does not modify but only reads it attributes
+	// and uses in other managed resources.
+	// +optional
+	Reference *NetworkReference `json:"reference,omitempty"`
+}
+
+// ManagedNetwork defines parameters for VPC network creation. In Azure and AWS networks must have CIDR address
+// space specified, while for GCP it can be empty.
+type ManagedNetwork struct {
+	// Cidr address range. For AWS and Azure a valid CIDR must be specified. For GCP can be empty.
+	Cidr string `json:"cidr,omitempty"`
+}
+
+// +kubebuilder:validation:MinProperties=1
+// +kubebuilder:validation:MaxProperties=1
+type NetworkReference struct {
+	// +optional
+	Gcp *GcpNetworkReference `json:"gcp,omitempty"`
+
+	// +optional
+	Azure *AzureNetworkReference `json:"azure,omitempty"`
+
+	// +optional
+	Aws *AwsNetworkReference `json:"aws,omitempty"`
+
+	// +optional
+	Openstack *OpenstackNetworkReference `json:"openstack,omitempty"`
+}
+
+type GcpNetworkReference struct {
+	GcpProject  string `json:"gcpProject"`
+	NetworkName string `json:"networkName"`
+}
+
+type AzureNetworkReference struct {
+	TenantId       string `json:"tenantId,omitempty"`
+	SubscriptionId string `json:"subscriptionId,omitempty"`
+	ResourceGroup  string `json:"resourceGroup"`
+	NetworkName    string `json:"networkName"`
+}
+
+type AwsNetworkReference struct {
+	AwsAccountId string `json:"awsAccountId"`
+	Region       string `json:"region"`
+	VpcId        string `json:"vpcId"`
+	NetworkName  string `json:"networkName"`
+}
+
+type OpenstackNetworkReference struct {
+	Domain      string `json:"domain"`
+	Project     string `json:"project"`
+	NetworkId   string `json:"networkId"`
+	NetworkName string `json:"networkName"`
+}
+
+// NetworkStatus defines the observed state of Network
+type NetworkStatus struct {
+	// +optional
+	State string `json:"state,omitempty"`
+
+	// List of status conditions to indicate the status of a Peering.
+	// +optional
+	// +listType=map
+	// +listMapKey=type
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+	// +optional
+	Network *NetworkReference `json:"network,omitempty"`
```

Comment:
Is this circular ref?

Reply:
No it's not. Check the class and object diagrams below

![class diagram](https://github.com/user-attachments/assets/4979f3bd-5358-4ee2-b618-4e29e3bf0b93)
![object diagram 1](https://github.com/user-attachments/assets/f2b60e54-4929-41fb-8b2e-4f30177bf72c)
![object diagram 2](https://github.com/user-attachments/assets/ca77514e-79df-44b7-8d6a-34d96a414290)